README.md

```diff
@@ -1,7 +1,111 @@
-### ReactiveFeedback
+# ReactiveFeedback
 
 Unideraction reactive architecture. This is [ReactiveSwift](https://github.com/ReactiveCocoa/ReactiveSwift) implemetation of the [RxFeedback](https://github.com/kzaher/RxFeedback)
 
-### Documentation
+## Documentation
 
-* TBD
+![](diagrams/ReactiveFeedback.jpg)
+
+### Motivation
+
+Requirements for iOS apps have become huge. Our code has to manage a lot of state e.g server responses, cached data, UI state, routing etc. Some may say that Reactive Programming can help us a lot, but in wrong hands, it can make even more harm to your code base.
+
+The goal of this library to provide a simple and intuitive approach to design reactive state machines.
+
+### Core Concepts
+
+##### State 
+
+Is a single source of truth. It represents a state of your system. Usually, plane swift type (does not contain any ReactiveSwift primitives) to the point that it can be saved on disk. The only way to change a `State` is to emit an `Event`
+
+```swift
+enum State {
+    case loading
+    case loaded([Item])
+}
+```
+
+##### Event
+
+Represent all possible events that can happen in your system which can transition to a new value of the `State`
+
+```swift
+enum Event {
+    case startLoadingNextPage
+    case loaded([Movie])
+    case failed(Error)
+    case retry
+}
+```
+
+##### Reducer 
+
+Is a pure function of `(State, Event) -> State`. This is the only place where the `State` can be changed.
+
+```swift
+func reduce(state: State, event: Event) -> State {
+    switch event {
+    case .startLoadingNextPage:
+        return .paging(context: state.context)
+    case .response(let batch):
+        var copy = state.context
+        copy.batch = batch
+        copy.movies += batch.results
+        return .loadedPage(context: copy)
+    case .failed(let error):
+        return .error(error: error, context: state.context)
+    case .retry:
+        return .retry(context: state.context)
+    }
+}
+```
+
+##### Feedback
+
+Represents effects that may happen in your system that somehow can mutate the `State`, e.g (UI events, Networking, DB fetches, timers, Bluetooth ...). Essentially it's a pure function of `Signal<State, NoError> -> Signal<Event, NoError>`
```

Comment:
> that somehow can mutate the `State`

Should mention that feebacks don't directly mutate the state, they only emit events, which cause the state to change in the reducer.